src/MIDebugEngine/AD7.Impl/AD7Thread.cs

```diff
@@ -276,26 +276,19 @@ int IDebugThread2.Resume(out uint suspendCount)
         // Sets the next statement to the given stack frame and code context.
         int IDebugThread2.SetNextStatement(IDebugStackFrame2 stackFrame, IDebugCodeContext2 codeContext)
         {
-            // CLRDBG TODO: This implementation should be changed to call an MI command
-            ulong addr = ((AD7MemoryAddress)codeContext).Address;
-            AD7StackFrame frame = ((AD7StackFrame)stackFrame);
-            if (frame.ThreadContext.Level != 0 || frame.Thread != this || !frame.ThreadContext.pc.HasValue || _engine.DebuggedProcess.MICommandFactory.Mode == MIMode.Clrdbg)
-            {
+            var infos = new CONTEXT_INFO[1];
+            if (codeContext.GetInfo(enum_CONTEXT_INFO_FIELDS.CIF_ADDRESS, infos) != Constants.S_OK)
                 return Constants.S_FALSE;
```

Comment:
Internally code contexts (AD7MemoryAddress) have the address as a `ulong` already. Probably makes more sense to verify that the provided code context is an `AD7MemoryAddress` and use that.